processing/src/main/java/io/druid/segment/data/ColumnarFloats.java

```diff
@@ -49,6 +50,45 @@ public float getFloat()
         return ColumnarFloats.this.get(offset.getOffset());
       }
 
+      @Override
+      public boolean isNull()
+      {
+        return false;
+      }
+
+      @Override
+      public double getDouble(int offset)
+      {
+        return ColumnarFloats.this.get(offset);
+      }
+
+      @Override
+      public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+      {
+        inspector.visit("columnar", ColumnarFloats.this);
+        inspector.visit("offset", offset);
+      }
+    }
+    return new HistoricalFloatColumnSelector();
+  }
+
+  default ColumnValueSelector<Float> makeColumnValueSelector(ReadableOffset offset, ImmutableBitmap nullValueBitmap)
+  {
+    final boolean hasNulls = !nullValueBitmap.isEmpty();
```

Comment:
Please align the logic with double. And if there is just one method, it checks anyway and returns an optimized impl.

Reply:
done.